src/ArmTemplates/Extractor/EntityExtractors/BackendExtractor.cs

```diff
@@ -134,6 +134,27 @@ void SaveBackendApiParametersToCache()
                     {
                         backendTemplate.TypedResources.BackendNameParametersCache.Add(backendValidName, backendApiParameters);
                     }
+
+                    if (backendResource.Properties.Proxy != null)
+                    {
+                        var proxyUniqueId = $"{backendResource.Properties.Proxy.Url}_{backendResource.Properties.Proxy.Username}";
```

Comment:
this will not lead to NRE, but what if `url` or `username` (one of them) is `null`? Then your `proxyUniqueId` would equal something like `_dmkorolev` or `contoso.com_`. I am not sure how we can make it work - maybe replace it with `url` and `username` placeholders so that we will end up with `proxyurl_dmkorolev` and `contoso.com_proxyuser`. Is that needed? What do you think?

Reply:
Regarding the url: 
 If proxy is set, url is required field so it will be always filled with valid value. 
Regarding the uniqueId:
 Underlines will be removed from unique identifier while generating the parameter variable name.  Example: contosocom
In case the username is not provided it is empty string and _ as was mentioned will be removed, also there is always a possibility to populate by default (which could be the same value from different backend proxy -> leads to inconsistency however it is rare case of course.) 
I suggest to leave it as it without changing the approach of handling empty usernames, (since empty usernames are already covered in current approach)
Open to discussion :)

Reply:
agreed. Mostly I was asking about the `url` and since it is required - I dont see any major trouble here